### ISSUE TITLE
Add support for the XDG Base Directory Specification

### ISSUE THREAD

**Issue by mohkale** [markdownlint/markdownlint#376]
## Describe the Enhancement:

Move `~/.mdlrc` out of my home directory and have mdl work without needing extra configuration.

### Impacted Rules:

None.

## Describe the Need:

The kind of user who'll appreciate this is the general purpose user who runs `ls -la ~/` more than twice a week.

## Current Alternative

```sh
mv ~/.mdlrc $XDG_CONFIG_HOME/mdlrc
```

In `~/.bashrc`:

```sh
alias mdl='mdl -c "$XDG_CONFIG_HOME/mdlrc"'
```

In `~/.config/emacs/init.el`:
```lisp
(setq flycheck-markdown-mdl-style (expand-file-name "~/.config/mdlrc"))
```

## Can We Help You Implement This?:

I'd be more than happy to implement if there's a motivation to accept. I'll ask for help at that stage.

**Comment by jaymzh** [comment]
Yeah this would be a good change. I've done this in some other apps I maintain as well. However, if we do this we need to make it backwards compatible. So if the XDG location doesn't exist, then look in the original location. Bonus points if you _move_ the config. :)

**Comment by mohkale** [comment]
@jaymzh 

Would you be ok width adding the [XDG gem](https://github.com/rubyworks/xdg) or prefer creating a builtin solution? Normally I'd say **do it yourself** for something this simple but seeing as we're conforming to a spec it seems like relying on an external library for details that might change would be a better idea. Thoughts?

**Comment by mohkale** [comment]
@jaymzh 

I'd just like to clarify some precedence rules. Looks like with a default config_file value of `.mdlrc`, we check:

```
- $PWD/.mdlrc
- $PWD/../.mdlrc
- $PWD/../../.mdlrc
- ...
- /.mdlrc
- ~/.mdlrc
```

I was thinking we could just insert a check for XDG between `/.mdlrc` and `~/.mdlrc`. My initial worry is that if `~/.mdlrc` exists and $PWD is a subdirectory of the home directory then mdl will prefer `~/.mdlrc` to `$XDG_CONFIG_HOME/mdlrc`. So just to maintain consistency I think it makes more sense to prefer `~/.mdlrc` and only check `$XDG_CONFIG_HOME/mdlrc` after `~/.mdlrc`.
